opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -439,7 +442,11 @@ namespace Opm {
           typedef Y range_type;
           typedef typename X::field_type field_type;
 
+#if HAVE_MPI
           typedef Dune::OwnerOverlapCopyCommunication<int,int> communication_type;
+#else
+          typedef Dune::CollectiveCommunication<int> communication_type;
```

Comment:
Theses two types are not semantically related as they have totally different interfaces.